src/repositories/submission_repository.rs

```diff
@@ -1,37 +1,60 @@
-use scylla::CachingSession;
-
-use crate::http::requests::submission_request::SubmissionDTO;
-use crate::http::SomeError;
-use crate::models::submission::Submission;
-
-pub struct SubmissionRepository {
-    db: Arc<CachingSession>,
-}
-
-
-impl SubmissionRepository {
-    pub fn new(db: Arc<CachingSession>) -> Self {
-        SubmissionRepository {
-            db
-        }
-    }
-
-    pub async fn insert(&self, payload: &SubmissionDTO) -> actix_web::Result<Submission, SomeError> {
-        let submission = Submission::from_request(&payload);
-        submission.insert().execute(&self.db).await?;
-
-        Ok(submission)
-    }
-
-    pub async fn find_by_id(&self, id: Uuid) -> actix_web::Result<Submission, SomeError> {
-        let submission = Submission { id, ..Default::default() };
-        let submission = submission.find_by_partition_key().execute(&self.db).await?;
-        let submission = submission.try_collect().await?.pop().unwrap();
-
-        Ok(submission)
-    }
-}
\ No newline at end of file
+use std::sync::Arc;
+
+use crate::http::requests::submission_request::SubmissionDTO;
+use crate::models::submission::Submission;
+use anyhow::Result;
+use charybdis::types::Uuid;
+use scylla::prepared_statement::PreparedStatement;
+use scylla::Session;
+
+const SUBMISSION_INSERT_QUERY: &str = "INSERT INTO submissions (id, song_id, player_id, modifiers, score, difficulty, instrument, played_at) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+const SUBMISSION_FIND_QUERY: &str = "SELECT id, song_id, player_id, modifiers, score, difficulty, instrument, played_at FROM submissions WHERE id = ?";
+
+#[derive(Clone)]
+pub struct SubmissionRepository {
+    db: Arc<Session>,
+    insert_query: PreparedStatement,
+    find_query: PreparedStatement,
+}
+
+impl SubmissionRepository {
+    pub async fn new(db: Arc<Session>) -> Self {
+        let insert_query = db
+            .prepare(SUBMISSION_INSERT_QUERY)
+            .await
+            .expect("Failed to prepare query");
+        let find_query = db
+            .prepare(SUBMISSION_FIND_QUERY)
+            .await
+            .expect("Failed to prepare query");
+        SubmissionRepository {
+            db,
+            insert_query,
+            find_query,
+        }
+    }
+
+    pub async fn insert(&self, payload: SubmissionDTO) -> Result<Submission> {
+        let submission = Submission::from_request(payload);
+        self.db
+            .execute_unpaged(&self.insert_query, &submission)
+            .await
+            .expect("Failed to insert submission");
+
+        Ok(submission)
+    }
+
+    pub async fn find_by_id(&self, id: Uuid) -> Result<Option<Submission>> {
+        let response = self
+            .db
+            .execute_unpaged(&self.find_query, (id,))
+            .await
+            .expect("Failed to query submission");
+
+        let submission = response
+            .into_rows_result()?
+            .maybe_first_row::<Submission>()?;
+
+        Ok(submission)
+    }
```

Comment:
Yeah I don't know enough about Scylla to tell if such select can be unpaged or not.